README.adoc

```diff
@@ -36,30 +38,238 @@ include::https://raw.githubusercontent.com/spring-guides/getting-started-macros/
+The Spring Cloud GCP Core Boot starter is bundled with the Spring Cloud GCP Pub/Sub Boot starter.
 
-`src/main/java/hello/GreetingController.java`
+The GCP project ID is auto-configured from the GOOGLE_CLOUD_PROJECT environment variable, among
+https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L287[several other sources].
+The OAuth2 credentials are auto-configured from the
+https://developers.google.com/identity/protocols/application-default-credentials[GOOGLE_APPLICATION_CREDENTIALS]
+environment variable. If the https://cloud.google.com/sdk/[Google Cloud SDK] is installed, this
+environment variable is easily configured by running the `gcloud auth application-default login`
+command in the same process of the app, or a parent one.
+
+== Create an inbound channel adapter
+
+An inbound channel adapter listens to messages from a Google Cloud Pub/Sub subscription and sends
+them to a Spring channel in an application.
+
+Instantiating an inbound channel adapter requires a subscriber factory and the name of an existing
+subscription. The Spring Cloud GCP Pub/Sub Boot starter provides a pre-configured subscriber factory
+which you can simply inject as a method argument.
+
+`src/main/java/hello/PubSubApplication.java`
 [source,java]
 ----
-include::complete/src/main/java/hello/GreetingController.java[]
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=messageChannelAdapter]
 ----
 
-NOTE: The above example does not specify `GET` vs. `PUT`, `POST`, and so forth, because `@RequestMapping` maps all HTTP operations by default. Use `@RequestMapping(method=GET)` to narrow this mapping.
+After the channel adapter is instantiated, an output channel where the adapter sends the received
+messages to must be configured. This channel can be explicitly declared through a bean.
 
+`src/main/java/hello/PubSubApplication.java`
+[source,java]
+----
+include::/complete/src/main/java/hello/PubSubApplication.java[tag=pubsubInputChannel]
+----
 
-== Make the application executable
+The `"testSubscription"` subscription must be created before the application is run. New
+subscriptions can be created through the
+https://console.cloud.google.com/cloudpubsub/subscriptions[Google Cloud Console] or,
+programatically, with the `PubSubAdmin` class.
 
-Although it is possible to package this service as a traditional link:/understanding/WAR[WAR] file for deployment to an external application server, the simpler approach demonstrated below creates a standalone application. You package everything in a single, executable JAR file, driven by a good old Java `main()` method. Along the way, you use Spring's support for embedding the link:/understanding/Tomcat[Tomcat] servlet container as the HTTP runtime, instead of deploying to an external instance.
+The message acknowledgement mode is set in the adapter to automatic, by default. The example shows
```

Comment:
This paragraph is too verbose. "Message acknowledgement mode" => easier to say that the Adapter automatically acknowledges messages and this may be overridden as shown below.